libensemble/gen_classes/aposmm.py

```diff
@@ -0,0 +1,125 @@
+import copy
+from typing import List
+
+import numpy as np
+from numpy import typing as npt
+
+from libensemble.generators import PersistentGenInterfacer
+from libensemble.message_numbers import EVAL_GEN_TAG, PERSIS_STOP
+
+
+class APOSMM(PersistentGenInterfacer):
+    """
+    Standalone object-oriented APOSMM generator
+    """
+
+    def __init__(
+        self,
+        variables: dict,
+        objectives: dict,
+        History: npt.NDArray = [],
+        persis_info: dict = {},
+        gen_specs: dict = {},
+        libE_info: dict = {},
+        **kwargs,
```

Comment:
For libEnsemble arguments this is opaque when looking at this routine. Perhaps they should be written out as additional arguments. Then there's the question of whether you keep gen_specs as well.